### PR TITLE
Instance: Handle SELinux for VM LXD agent

### PR DESCRIPTION
**Pull request by MusicDin**
Cherry picked from https://github.com/lxc/incus/pull/489

### CODE REVIEW

**Comment by tomponline** [comment]
> @simondeziel does this need to go into the systemd units package too?

I suppose not as  Ubuntu doesn't use selinux right?

**Review by simondeziel — COMMENTED** [review]
Ubuntu supports using SELinux but it's not a pleasant experience I've heard. That said, those changes should indeed go in the lxd-agent-loader too.

**Comment by simondeziel** [comment]
> @simondeziel does this need to go into the systemd units package too?

https://code.launchpad.net/~sdeziel/ubuntu/+source/lxd-agent-loader/+git/lxd-agent-loader/+merge/460458

File: lxd/instance/drivers/driver_qemu.go [hunk 1]
```diff
@@ -2813,6 +2816,11 @@ cp systemd/lxd-agent.service /lib/systemd/system/
 cp systemd/lxd-agent-setup /lib/systemd/
 systemctl daemon-reload
 
+# SELinux handling.
+if getenforce >/dev/null 2>&1; then
+    semanage fcontext -a -t bin_t /var/run/lxd_agent/lxd-agent
```
[the remark this answers is not in the document]
Comment (MusicDin) [reply lxd/instance/drivers/driver_qemu.go]: Fixed. Thanks for review 👍

File: lxd/instance/drivers/driver_qemu.go [hunk 2]
```diff
@@ -2813,6 +2816,11 @@ cp systemd/lxd-agent.service /lib/systemd/system/
 cp systemd/lxd-agent-setup /lib/systemd/
 systemctl daemon-reload
 
+# SELinux handling.
+if getenforce >/dev/null 2>&1; then
+    semanage fcontext -a -t bin_t /run/lxd_agent/lxd-agent
```
Comment (tomponline) [inline lxd/instance/drivers/driver_qemu.go:2821]: And is `/run/lxd_agent/lxd-agent` where the lxd-agent is actually placed?

Comment (MusicDin) [reply lxd/instance/drivers/driver_qemu.go:2821]: Yes, it is placed in `/run/lxd_agent.lxd-agent`
```
$ ls -lah /run/lxd_agent/lxd-agent 
-r-x------ 1 root root 17M Feb 13 14:11 /run/lxd_agent/lxd-agent
```

As @simondeziel pointed out, on this path we mount the agent in the [setup script](https://github.com/canonical/lxd/blob/main/lxd/instance/drivers/driver_qemu.go#L2731-L2733), and [systemd unit](https://github.com/canonical/lxd/blob/main/lxd/instance/drivers/driver_qemu.go#L2706-L2721) also uses that path.

**Review by tomponline — APPROVED** [review]

**Pull request merged**